server/athenian/api/internal/settings.py

```diff
@@ -638,6 +638,27 @@ def repo_exists(self, logical_repo: RepositoryName) -> bool:
             return False
         return logical_repo.unprefixed in physical_repo_settings.logical_repositories
 
+    def explode_user_repos(self, repos: Iterable[str]) -> list[str]:
+        """Explode the list of repositories requested by user.
+
+        Each physical repo is exploded with its logical components if none of them is in the list.
```

Comment:
```suggestion
        We insert all logical components of a repo if at least one logical component appears in `repos`.
```
(that's my understanding of the logic)

Reply:
That's the opposite, if no logical components is in the list the user wants to refer everything so we add all logical components. 
https://athenian.slack.com/archives/CQU6FR09Z/p1672665741875189?thread_ts=1672664721.132099&cid=CQU6FR09Z
Given repos `athenianco/r0` and `athenianco/r0/l0` `athenianco/r0/l1`logical components:
`[athenianco/r0] => [atheniaco/r0, athenianco/r0/l0, athenianco/r0/l1]`
`[atheniaco/r0, athenianco/r0/l0] => [atheniaco/r0, athenianco/r0/l0]`

Reply:
I've rephrased the docstring